packages/hash/api/src/lib/error.ts

```diff
@@ -0,0 +1,6 @@
+export class TypeMismatchError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "TypeMismatchError";
+  }
```

Comment:
Maybe something like this?
```suggestion
export class EntityTypeMismatchError extends Error {
  constructor(entityId: entityId, expectedTypeId: VersionedUri, actualTypeId: VersionedUri) {
    super(`expected entity with id "${entityId}" to be of type "${expectedTypeId}" but got:  ${actualTypeId}`);
    this.name = "TypeMismatchError";
    this.expectedTypeId = expectedTypeId;
    this.actualTypeId = actualTypeId;
  }
```

Then in the handler you could do `error instanceof TypeMismatchError && error.actualTypeId === WORKSPACE_TYPES.entityType.org.schema.$id`

Reply:
Thanks for the suggestion, agree this would be much better :)

Have addressed in https://github.com/hashintel/hash/pull/1324/commits/2f043d500d96f5e2bc697f6833a18eb44fbf3d73